### PR TITLE
feat add input

### PR DESCRIPTION
**Pull request by Sebastien-Ahkrin**
Parts of: https://github.com/zakodium-oss/analysis-ui-components/issues/346

- [x] Controlled component
- [x] Forwards the native input props
- [x] Placeholder
- [x] 2 sizes
- [x] With / without labels
- [x] Size on Field too
- [x] trailing / leading component (inline or not)
- [ ] validation feedback
- [x] handle required props

### CODE REVIEW

File: stories/components/input.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { ChangeEvent, useState } from 'react';
+
+import { Input } from '../../src/components/forms/Input';
+import {
+  Field,
+  Fields,
+} from '../../src/components/forms/context/FieldsContext';
```
Comment (targos) [inline stories/components/input.stories.tsx:7]: all should be imported from `../../src/components/index`

**Comment by stropitek** [comment]
You can add the `required` prop to the `Field` component? There is an example in the issue of how a required field can look like.

**Comment by stropitek** [comment]
- Story with small and regular sizes next to each other
- Story with small and regular sizes next to each other but with label (should work by setting a `size` prop to `Field` not `Input`

File: src/components/forms/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  min-width: 0px;
+  background-color: #fff;
+  border-radius: ${(props) => (props.inputSize === 'default' ? '6px' : '4px')};
+  transition: all 0.2s;
+  border: 1px solid #d9d9d9;
+`;
+
+export interface InputProps
+  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'size'> {
+  inputSize?: 'default' | 'small';
```
Comment (stropitek) [inline src/components/forms/Input.tsx:25]: I neither like highjacking the `size` prop nor the `inputSize` name. Maybe call it variant?

**Comment by Sebastien-Ahkrin** [comment]
Do i have to add a "size" props on the FieldContext for Input @stropitek  ?

**Comment by stropitek** [comment]
> Do i have to add a "size" props on the FieldContext for Input

yep. If size set directly on input it takes precedence

**Comment by stropitek** [comment]
@Sebastien-Ahkrin what is the state of this PR?

**Comment by Sebastien-Ahkrin** [comment]
> @Sebastien-Ahkrin what is the state of this PR?

@stropitek i'm working on Leading & Trailing addon actually, but actually i can make the pull request okay to check & merge. And i finalize my work on Trailing on a new pull request

**Comment by stropitek** [comment]
Did you already work on the latest feedback that was given (no commits after them)?



> i'm working on Leading & Trailing addon actually

Yes please don't include that in this PR

**Comment by stropitek** [comment]
image

On Edge (the browser) there is the default black outline. It should be removed.

**Comment by stropitek** [comment]
If the size prop is not used to change the appearance of the label, then there is no need to have it on `Field`

**Comment by stropitek** [comment]
The outline on the input should stay when the input is focused

**Comment by stropitek** [comment]
The label should have the same font size as the input.

Therefore I think it makes sense to add the `variant` prop to the `Field` component. It can be inherited by the input via the context.

File: src/components/forms/Input.tsx [hunk 3]
```diff
@@ -133,7 +133,7 @@ export function Input(props: InputProps) {
       <InputStyled
         id={name}
         name={name}
-        variant={variant}
+        variant={contextVariant || variant}
```
Comment (stropitek) [inline src/components/forms/Input.tsx:136]: It should be `variant` || `contextVariant` and `variant` should not have a default value, but `contextVariant` should

**Comment by stropitek** [comment]
image

**Comment by Sebastien-Ahkrin** [comment]
> image

@stropitek what navigator are you using and what's the name of the stories for this screen ?

File: src/components/forms/Input.tsx [hunk 4]
```diff
@@ -24,7 +24,13 @@ const GroupStyled = styled.div`
   }
 `;
 
-const lineHeight = 4 / 7;
+function getSpecialSize(props: any) {
```
Comment (stropitek) [inline src/components/forms/Input.tsx:27]: 👀

**Review by stropitek — APPROVED** [review]

**Pull request merged**